File: Portal SAE/backend/src/main/java/com/siscem/portal_sae/dtos/task/TaskDTO.java
package com.siscem.portal_sae.dtos.task;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Data;

@Data
public class TaskDTO {
	private Integer codigo;
	private String asunto;
	@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
	private LocalDateTime fechaHoraInicio;
	@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
	private LocalDateTime fechaHoraFin;
	private String porcentajeProgreso;
	private String descripcion;
	private String observacion;

	// Foreign Keys
	private String consultorAsignado;
	private String empresarioAsignado;
	private String cuentaAsignada;
	private String estado;
	private String tipo;
}

File: Portal SAE/frontend/src/app/interfaces/task/task.dto.ts
export interface TaskDTO {
  codigo: number;
  asunto: string;
  fechaHoraInicio: string;
  fechaHoraFin: string;
  porcentajeProgreso: string;
  descripcion: string;
  observacion: string;

  consultorAsignado: string;
  empresarioAsignado: string;
  cuentaAsignada: string;
  estado: string;
  tipo: string;
}
